### PR TITLE
Add Layouts via AddElementMenu

### PR DESCRIPTION
**Pull request by oshi97**
This PR implements adding layouts via the AddElementMenu/Button.

J=SLAP-2924
TEST=manual,auto

I can add layouts multiple times, set the newly added components as active, and save to file
updated AddElementMenu tests

### CODE REVIEW

File: e2e-tests/tests/infra/StudioPlaywrightPage.ts [hunk 1]
```diff
@@ -183,15 +183,15 @@ export default class StudioPlaywrightPage {
 
   async addElement(
     elementName: string,
-    category: "Components" | "Layouts",
+    category: "Components" | "Containers",
```
Comment (tmeyer2115) [inline e2e-tests/tests/infra/StudioPlaywrightPage.ts:186]: Should we also include "Layouts" in this union type?

File: e2e-tests/tests/infra/StudioPlaywrightPage.ts [hunk 2]
```diff
@@ -183,15 +183,15 @@ export default class StudioPlaywrightPage {
...
     shouldTakeScreenshots = true
   ) {
     await this.openAddElementMenu(category, shouldTakeScreenshots);
     await this.getAddElementLocator(elementName).click();
   }
 
   async openAddElementMenu(
-    category: "Components" | "Layouts" = "Components",
+    category: "Components" | "Containers" = "Components",
```
Comment (tmeyer2115) [inline e2e-tests/tests/infra/StudioPlaywrightPage.ts:194]: Same comment here.

File: packages/studio/src/components/common/renderIconForType.tsx [hunk 3]
```diff
@@ -10,6 +10,7 @@ export default function renderIconForType(type: ElementType) {
     case ElementType.Components:
       return <Box />;
     case ElementType.Containers:
+    case ElementType.Layouts:
```
Comment (tmeyer2115) [inline packages/studio/src/components/common/renderIconForType.tsx:13]: Can we ask Product if there's an icon for Layouts? If there isn't, let's make sure they're cool with re-using the Container icon.

File: packages/studio/src/store/slices/createLayoutSlice.ts [hunk 4]
```diff
@@ -4,7 +4,10 @@ import { SliceCreator } from "../models/utils";
 import removeTopLevelFragments from "../../utils/removeTopLevelFragments";
 
 const createLayoutSlice: SliceCreator<LayoutSlice> = () => ({
-  layouts: removeTopLevelFragments(initialStudioData.layoutNameToLayoutState),
+  // TODO (SLAP-2930): Remove top-level fragments from layouts
```
Comment (tmeyer2115) [inline packages/studio/src/store/slices/createLayoutSlice.ts:7]: Do we need this TODO any more? Seems like we are removing the top-level fragments from layouts.

File: packages/studio/src/store/StudioActions.ts [hunk 5]
```diff
@@ -183,6 +185,26 @@ export default class StudioActions {
...
+            ...componentState,
+            uuid: v4(),
+          };
+          if (parentState) {
+            updatedComponentState.parentUUID = parentState.uuid;
+          }
+          return updatedComponentState;
+        }
+      );
+    const updatedTree = layoutTreeWithNewUUIDs.concat(tree);
```
Comment (alextaing) [inline packages/studio/src/store/StudioActions.ts:204]: I recall overhearing at some point that we were planning on placing the new layout at the end of the component tree-- did product decide on adding it to the beginning?

File: packages/studio/src/components/common/renderIconForType.tsx (hunk 3 above)
Comment (oshi97) [reply packages/studio/src/components/common/renderIconForType.tsx:13]: I pinged Aaron!

Comment (oshi97) [reply packages/studio/src/components/common/renderIconForType.tsx:13]: He said he's not sure, probably we can merge this in and update the icon later?

File: packages/studio/src/store/StudioActions.ts (hunk 5 above)
Comment (oshi97) [reply packages/studio/src/store/StudioActions.ts:204]: Aaron says top!

File: packages/studio/src/components/common/renderIconForType.tsx (hunk 3 above)
Comment (tmeyer2115) [reply packages/studio/src/components/common/renderIconForType.tsx:13]: Sounds good!

**Review by tmeyer2115 — APPROVED** [review]

**Pull request merged**